### PR TITLE
accepts Rack::Test::UploadFile for specs

### PR DESCRIPTION
**Pull request by PositiveControl**

### CODE REVIEW

File: lib/teamsnap.rb [hunk 1]
```diff
@@ -68,7 +68,9 @@ def client_send(client, via, href, args)
         client.send(via, href, args)
       when :patch, :post
         client.send(via, href) do |req|
-          if args.values.any? { |a| a.kind_of?(File) || a.kind_of?(Tempfile) }
+          if valid_file_types.any? { |type|
+            args.values.any? { |a| defined?(type) && a.kind_of?(type) }
+          }
```
Comment (glanotte) [inline lib/teamsnap.rb]: what about extracting this to a method?

Comment (glanotte) [inline lib/teamsnap.rb]: something like `use_multipart?(type)`

Comment (PositiveControl) [inline lib/teamsnap.rb]: Great idea... I like that.  Coming right up!

**Comment by glanotte** [comment]
⛵

**Pull request merged**